llvm/lib/TableGen/TGLexer.cpp

```diff
@@ -656,17 +654,13 @@ tgtok::TokKind TGLexer::LexExclaim() {
 bool TGLexer::prepExitInclude(bool IncludeStackMustBeEmpty) {
   // Report an error, if preprocessor control stack for the current
   // file is not empty.
-  if (!PrepIncludeStack.back()->empty()) {
+  if (!PrepIncludeStack.back().empty()) {
     prepReportPreprocessorStackError();
 
     return false;
   }
 
   // Pop the preprocessing controls from the include stack.
-  if (PrepIncludeStack.empty()) {
```

Comment:
This check was pointless since we have already accessed `PrepIncludeStack.back()` above.

Reply:
should we instead move this check before accessing `PrepIncludeStack.back()` above? I feel like an error message will be more helpful than simply crash in the case of PrepIncludeStack being empty.

Reply:
I don't think any valid input could cause the stack to be empty here, so this should just be an assertion, but `.back()` should already assert for us.